src/roman_datamodels/maker_utils/_datamodels.py

```diff
@@ -409,6 +411,201 @@ def mk_guidewindow(*, shape=(2, 8, 16, 32, 32), filepath=None, **kwargs):
+    )
+    l1detectorgwcentroid["acq_centroid_times"] = kwargs.get(
+        "acq_centroid_times", [time.Time("2024-01-01T12:00:00", format="isot", scale="utc")] * shape[0]
+    )
+    l1detectorgwcentroid["track_centroids"] = kwargs.get("track_centroids", np.zeros(shape[1:], dtype=np.float32))
+    l1detectorgwcentroid["track_centroid_errs"] = kwargs.get("track_centroid_errs", np.zeros(shape[1:], dtype=np.float32))
+    l1detectorgwcentroid["track_centroid_quality"] = kwargs.get("track_centroid_quality", np.array(["?"] * shape[0], dtype="<U7"))
+    l1detectorgwcentroid["track_centroid_times"] = kwargs.get(
+        "track_centroid_times", [time.Time("2024-01-01T12:00:00", format="isot", scale="utc")] * shape[0]
+    )
+
+    if mode == "WSM":
+        l1detectorgwcentroid["edge_acq_centroids"] = kwargs.get("edge_acq_centroids", np.zeros(shape[1:], dtype=np.float32))
+        l1detectorgwcentroid["edge_acq_centroid_errs"] = kwargs.get(
+            "edge_acq_centroid_errs", np.zeros(shape[1:], dtype=np.float32)
+        )
+        l1detectorgwcentroid["edge_acq_centroid_quality"] = kwargs.get(
+            "edge_acq_centroid_quality", np.array(["?"] * shape[0], dtype="<U7")
+        )
+        l1detectorgwcentroid["edge_acq_centroid_times"] = kwargs.get(
+            "edge_acq_centroid_times", [time.Time("2024-01-01T12:00:00", format="isot", scale="utc")] * shape[0]
+        )
+
+    return l1detectorgwcentroid
+
+
+def mk_l1_detector_guidewindow_array(*, name, shape=(16, 32, 32), **kwargs):
+    """
+    Create a dummy Level 1 detector guidewindow data arrays instance (or file) with
+    arrays and valid values for attributes required by the schema.
+
+    Parameters
+    ----------
+    name
+        Name of array block to create.
+
+    shape
+        (optional, keyword-only) Shape of arrays in the model.
+
+    filepath
+        (optional, keyword-only) File name and path to write model to.
+
+    Returns
+    -------
+    dict (defined by the l1_detector_guidewindow-1.0.0 schema with additional guidewindow metadata)
+    """
+    if len(shape) != 3:
+        shape = (16, 32, 32)
+        warnings.warn("Input shape must be 3D. Defaulting to (16, 32, 32)", UserWarning, stacklevel=2)
+
+    l1detectorgwarr = {}
+
+    l1detectorgwarr["pedestal_resultants"] = kwargs.get("pedestal_resultants", np.zeros(shape, dtype=np.uint16))
+    l1detectorgwarr["signal_resultants"] = kwargs.get("signal_resultants", np.zeros(shape, dtype=np.uint16))
+    l1detectorgwarr["lower_left_corners"] = kwargs.get("lower_left_corners", np.zeros(shape[1:], dtype=np.uint16))
+    l1detectorgwarr["upper_right_corners"] = kwargs.get("upper_right_corners", np.zeros(shape[1:], dtype=np.uint16))
+    l1detectorgwarr["pixel_offsets"] = kwargs.get("pixel_offsets", np.zeros(shape[0], dtype=np.uint16))
+    l1detectorgwarr["reset_impacted_pairs"] = kwargs.get("reset_impacted_pairs", np.zeros(shape[0], dtype=np.bool))
```

Comment:
```suggestion
    l1detectorgwarr["reset_impacted_pairs"] = kwargs.get("reset_impacted_pairs", np.zeros(shape[0], dtype=bool))
```
To prevent the oldest-deps failure.